test/test_field.jl

```diff
@@ -451,4 +450,67 @@ end
             end
         end
     end
+
+    @testset "Views of field views" begin
+        @info "  Testing views of field views..."
+
+        FieldTypes = (CenterField, XFaceField, YFaceField, ZFaceField)
+        ZTopologies = (Periodic, Bounded)
+
+        for arch in archs, FT in float_types, FieldType in FieldTypes, ZTopology in ZTopologies

```

Comment:
I know its written all over the tests, from a distance of a few years of writing this code I wonder if something like this would be easier to understand:

```suggestion
        for n = 1:2
        
            arch = archs[n]
            FT = float_types[n]
            FieldType = FieldTypes[n]
            ZTopology = ZTopologies[n]
```

Reply:
I don't really like that that tbh

Reply:
Then we have to write something like

```julia
for arch in archs, FT in float_types, FieldType in FieldTypes, (ZTopology, k_top) in zip(ZTopologies, k_tops)
```

Reply:
I realized what I wrote above is not equivalent to the current code, which does every permutation (not just 2 cases, its 2^4 cases)

Reply:
Even the code snippet
```julia
        Nx, Ny, Nz = 1, 1, 7

        FieldTypes = (CenterField, XFaceField, YFaceField, ZFaceField)
        ZTopologies = (Periodic, Bounded)
        k_tops = (Nz, Nz+1)

        for arch in archs, FT in float_types, FieldType in FieldTypes, (ZTopology, k_top) in zip(ZTopologies, k_tops)
            grid = RectilinearGrid(arch, FT, size=(Nx, Ny, Nz), x=(0, 1), y=(0, 1), z=(0, 1), topology = (Periodic, Periodic, ZTopology))
            Hx, Hy, Hz = halo_size(grid)
```
would yield erroneous results since `k_top = Nz + 1` only for a `ZFaceField` with bounded topology in `z` and `k_top = Nz` for all other cases. So, I adjusted the implementation as follows:
```julia
        Nx, Ny, Nz = 1, 1, 7

        FieldTypes = (CenterField, XFaceField, YFaceField, ZFaceField)
        ZTopologies = (Periodic, Bounded)

        for arch in archs, FT in float_types, FieldType in FieldTypes, ZTopology in ZTopologies
            grid = RectilinearGrid(arch, FT, size=(Nx, Ny, Nz), x=(0, 1), y=(0, 1), z=(0, 1), topology = (Periodic, Periodic, ZTopology))
            Hx, Hy, Hz = halo_size(grid)

            k_top = (FieldType == ZFaceField && ZTopology == Bounded) ? Nz+1 : Nz 
```

Reply:
great